Code/Framework/AzCore/AzCore/Settings/ConfigurableStack.h

```diff
@@ -0,0 +1,189 @@
+    //!         {
+    //!             "MyValue": "hello",
+    //!             "ExampleValue": 84
+    //!         },
+    //!         "TheSecondEntry":
+    //!         {
+    //!             "MyValue": "world"
+    //!         }
+    //!     }
+    //! }
+    //! 
+    //! // Customized settings.
+    //! {
+    //!     "Settings in a stack":
+    //!     {
+    //!         // Add a new entry before "AnOriginalEntry" in the original document.
+    //!         "NewEntry":
+    //!         {
+    //!             "$stack_before": "AnOriginalEntry",
+    //!             "MyValue": 42
+    //!         },
+    //!         // Add a second entry after "AnOriginalEntry" in the original document.
+    //!         "SecondNewEntry":
+    //!         {
+    //!             "$stack_after": "AnOriginalEntry",
+    //!             "MyValue": "FortyTwo".
+    //!         },
+    //!         // Update a value in "AnOriginalEntry".
+    //!         "AnOriginalEntry":
+    //!         {
+    //!             "ExampleValue": 42
+    //!         },
+    //!         // Delete the "TheSecondEntry" from the settings.
+    //!         "TheSecondEntry" : null, 
+    //!     }
+    //! }
+    //!
+    //! The ConfigurableStack uses an AZStd::shared_ptr to store the values. This supports settings up a base class and specifying
+    //! derived classes in the settings, but requires that the base and derived classes all have a memory allocator associated with
+    //! them (i.e. by using the "AZ_CLASS_ALLOCATOR" macro) and that the relation of the classes is reflected. Loading a
+    //! ConfigurableStack can be done using the GetObject call on the SettingsRegistryInterface.
+
+    class ReflectContext;
+
+    class ConfigurableStackInterface
+    {
+    public:
+        friend class JsonConfigurableStackSerializer;
+
+        virtual ~ConfigurableStackInterface() = default;
+
+        virtual const TypeId& GetNodeType() const = 0;
+
+    protected:
+        enum class InsertPosition
+        {
+            Before,
+            After
+        };
+        virtual void* AddNode(AZStd::string name) = 0;
+        virtual void* AddNode(AZStd::string name, AZStd::string_view target, InsertPosition position) = 0;
+    };
+
+    template<typename StackBaseType>
+    class ConfigurableStack final : public ConfigurableStackInterface
+    {
+    public:
+        using NodeValue = AZStd::shared_ptr<StackBaseType>;
+        using Node = AZStd::pair<AZStd::string, NodeValue>;
+        using NodeContainer = AZStd::vector<Node>;
+        using Iterator = typename NodeContainer::iterator;
+        using ConstIterator = typename NodeContainer::const_iterator;
+
+        ~ConfigurableStack() override = default;
+
+        const TypeId& GetNodeType() const override;
+
+        Iterator begin();
+        Iterator end();
+        ConstIterator begin() const;
+        ConstIterator end() const;
+        ConstIterator cbegin() const;
+        ConstIterator cend() const;
```

Comment:
Actually with the new range support. The need to implement cbegin and cend aren't needed.

[std::ranges::cbegin](https://en.cppreference.com/w/cpp/ranges/cbegin) and [std::ranges::cend](https://en.cppreference.com/w/cpp/ranges/cend) will use the `begin` and `end` const member functions if available

Reply:
Will remove them then as less code to maintain is better.